client/extensions/woocommerce/app/order/index.js

```diff
@@ -54,14 +54,14 @@ class Order extends Component {
 	}
 
 	render() {
-		const { className, isSaving, order, site, translate } = this.props;
+		const { className, isSaving, order, orderId, site, translate } = this.props;
 		if ( ! order ) {
 			return null;
 		}
 
 		const breadcrumbs = [
 			( <a href={ getLink( '/store/orders/:site/', site ) }>{ translate( 'Orders' ) }</a> ),
-			( <span>{ translate( 'Order Details' ) }</span> ),
+			( <span>{ translate( 'Order Details %(orderId)s', { args: { orderId: `#${ orderId }` } } ) }</span> ),
```

Comment:
Not really something I thought hard about, but it's not translatable in the orders table, so I didn't think it was necessary here. Do you think that should change? I can wrap translation around the `#{orderId}` in the list table too.